docs/integrations/data-ingestion/clickpipes/postgres/ordering_keys.md

```diff
@@ -0,0 +1,113 @@
+
+For example, in a multi-tenant SaaS application, using (`tenant_id`, `id`) as the ordering key is a good choice. These columns uniquely identify each row, and `tenant_id` remains constant for an `id` even if other columns change. Since deduplication by id aligns with deduplication by (tenant_id, id), it helps avoid data [deduplication issues](https://docs.peerdb.io/mirror/ordering-key-different) that could arise if tenant_id were to change.
+
+**Note**: If you have scenarios where ordering keys need to include columns that change, please reach out to us at support@clickhouse.com. There are advanced methods to handle this, and we will work with you to find a solution.
+
+**Set Replica Identity on Postgres Tables to Custom Ordering Key**
+
+For Postgres CDC to function as expected, it is important to modify the `REPLICA IDENTITY` on tables to include the ordering key columns. This is essential for handling DELETEs accurately.
+
+If the `REPLICA IDENTITY` does not include the ordering key columns, Postgres CDC will not capture the values of columns other than the primary key - this is a limitation of Postgres logical decoding. All ordering key columns besides the primary key in Postgres will have nulls. This affects deduplication, meaning the previous version of the row may not be deduplicated with the latest deleted version (where `_peerdb_is_deleted` is set to 1).
+
+In the above example with `owneruserid` and `id`, if the primary key does not already include `owneruserid`, you need to have a `UNIQUE INDEX` on (`owneruserid`, `id`) and set it as the `REPLICA IDENTITY` for the table. This ensures that Postgres CDC captures the necessary column values for accurate replication and deduplication.
+
+Below is an example of how to do this on the events table. Make sure to apply this to all tables with modified ordering keys.
+
+```sql
+-- Create a UNIQUE INDEX on (owneruserid, id)
+CREATE UNIQUE INDEX posts_unique_owneruserid_idx ON posts(owneruserid, id);
+-- Set REPLICA IDENTITY to use this index
+ALTER TABLE posts REPLICA IDENTITY USING INDEX posts_unique_owneruserid_idx;
+```
+
+### Projections {#projections}
+
+[Projections](/sql-reference/statements/alter/projection) are useful for running queries on a column that is not a part of the primary key.
+
+> The biggest caveat with Projections is that they get skipped when querying the table with the FINAL keyword and do not account for deduplication. This could work for a few use cases where duplicates (updates, deletes) are not present or are less common.
+
+Projections are defined on the table we want to add a custom ordering key for. Then, each time a query is executed on this table, ClickHouse determines if the query execution can benefit from using one of the existing Projections.
+
+Let's take an example where we want to order the table posts by the field `creationdate` instead of the current one id. This would benefit query that filter using a date range.
```

Comment:
```suggestion
Let's take an example where we want to order the table posts by the field `creationdate` instead of the current one, `id`. This would benefit queries that filter using a date range.
```